server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1069,7 +1071,8 @@ private DefaultSearchContext createSearchContext(
                 minimumDocsPerSlice,
                 fetchPhase,
                 lowLevelCancellation,
-                this.enableConcurrentCollection && concurrentSearchEnabled(resultsType, request.source())
+                this.enableConcurrentCollection ? concurrentSearchExecutor : null,
```

Comment:
It seems weird to me that we now have two settings, `enableConcurrentCollection` set by `SEARCH_CONCURRENCY_ENABLED` and `SEARCH_CONCURRENT.size == 1`. 

Is it worth it to set `SEARCH_CONCURRENCY_ENABLED` when a user can set ``SEARCH_CONCURRENT.size = 1`?

Reply:
There is a subtle difference: one entirely disables concurrency, meaning the concurrent thread pool is no longer used. That is an escape hatch that is currently undocumented.

The force single slice one instead depends on the phase and the request, and forces the generation of a single slice, which though will be offloaded to the concurrent thread pool. 

This distinction is important internally to make sure that once we execute concurrently either one of the thread pools does the heavy work, otherwise both may do it and that could overload the system and make sizing more difficult.

Reply:
I simplified this: we were already providing a max num slices, I can just set that to 1 instead of requiring an additional forceSingleSlice argument.